### PR TITLE
Added blueprint arugment 

### PR DESCRIPTION
**Pull request by andytai7**
# Description
Added blueprint argument so that if blueprint already exists, this step is skipped. 

> Add directions for contributors. For example, do developers need to run `poetry install` after picking up these changes?

# Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Maintenance
- [ ] New release

# Related issues
> Mention related GitHub and Linear issues. E.g. Closes #xxx or Fixes #xxx. Otherwise delete this section.

# Checklists
To speed up the review process, please follow these checklists:

## Development
- [X] The Pull Request is small and focused on one topic
- [ ] Lint rules pass locally (`make format && make lint`)
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development (`make test`)
- [ ] The changes generate no new warnings (or explain any new warnings and why they're ok)
- [X] Commit messages are detailed
- [ ] Changed code is self-explanatory and/or I added comments
- [ ] I updated the documentation (docstrings, /docs)
See the testing guidelines for help on tests, especially those involving web services.

## Code review
- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [ ] I have performed a self-review of my code
- [ ] Issue from task tracker has a link to this pull request

💔 Thank you for submitting a pull request!

### CODE REVIEW

**Review by amirfz — CHANGES_REQUESTED** [review]
@andytai7 could you please also update the tutorial with this information? I'm esp interested in this: if I want to provide an external blueprint, what format does it have to have? it might even be best to include a template for it that I can download, modify, and use.

**Comment by amirfz** [comment]
@20001LastOrder looks good to me. could you review and merge if it looks good?

File: demo/blog_writer/main.py [hunk 1]
```diff
@@ -62,30 +61,47 @@ def get_user_agent_from_config_file(
     parser = ArgumentParser()
     parser.add_argument("--config", type=str, default="agent_config.yaml")
     parser.add_argument("--transcript", type=str, default="transcript.txt")
+    parser.add_argument("--blueprint", type=str, help="Optional JSON blueprint file to use")
```
Comment (20001LastOrder) [inline demo/blog_writer/main.py:64]: By default, arguments in `add_argument` are required. You will need to add a default value such as `default=None` to make it optional I think.

Comment (20001LastOrder) [reply demo/blog_writer/main.py:64]: Nevermind, I think this is no longer the case.

**Comment by 20001LastOrder** [comment]
@amirfz This looks good to me. But you will need to approve it before I can merge (since you asked for some changes before)

**Review by amirfz — APPROVED** [review]

**Pull request merged**